index.js

```diff
@@ -16,35 +17,64 @@ module.exports = ({utPort, registerErrors, utBus}) => class CachePort extends ut
     init(...params) {
         Object.assign(this.errors, registerErrors(errors));
         const {engine, options} = this.config.client;
-        this.client = new Catbox.Client(engine, options);
+        this.client = new Catbox.Client(engine || require('catbox-memory'), options);
         return super.init(...params);
     }
     async start(...params) {
         await this.client.start();
-        this.pull((msg = {}, $meta = {}) => {
-            const method = $meta.method;
-            if (!method) throw utBus.errors['bus.missingMethod']();
-            const handler = this.methods[method];
-            if (typeof handler !== 'function') throw utBus.errors['bus.methodNotFound']({params: {method}});
-            return handler(msg, $meta).catch(e => {
-                throw this.errors.cache(e);
-            });
-        });
+        this.pull(this.exec);
         return super.start(...params);
     }
-    handlers() {
-        const handlers = {
-            'cache.get': ({id, segment}) => this.client.get({id, segment}),
-            'cache.set': ({id, segment, value, ttl = 0}) => this.client.set({id, segment}, value, ttl),
-            'cache.drop': ({id, segment}) => this.client.drop({id, segment})
+    async exec(...params) {
+        let $meta = params && params.length > 1 && params[params.length - 1];
+        let methodName = $meta && $meta.method;
+        if (!methodName) throw utBus.errors['bus.missingMethod']();
+        let cache = $meta && $meta.cache;
+        if (!cache) throw this.errors['cachePort.missingCache']({methodName});
+        if (!['get', 'set', 'drop'].includes(cache.operation)) throw this.errors['cachePort.missingOperation']({methodName, operation: cache.operation});
+        let segment = cache.key && cache.key.segment;
+        if (!segment) {
+            let cacheParams = cache.key && cache.key.params;
+            if (cacheParams && typeof cacheParams === 'object') {
+                cacheParams = new url.URLSearchParams(cacheParams);
+                cacheParams.sort();
+            }
+            segment = cacheParams ? methodName + '?' + cacheParams.toString() : methodName;
+        }
+        let id = cache.key && cache.key.id;
+        if (id == null) {
+            throw this.errors['cachePort.missingId']({methodName});
+        }
+        try {
+            let fn = this.methods[segment] || this;
+            let cached = await fn[cache.operation].call(this, {id, segment, value: params[0], ttl: cache.ttl});
+            return cached && cached.item;
+        } catch (e) {
+            throw this.errors.cachePort(e);
         };
+    }
+    get({id, segment}) {
+        return this.client.get({id, segment});
+    }
+    set({id, segment, value, ttl = this.config.ttl}) {
+        this.client.set({id, segment}, value, ttl);
+        return null;
+    }
+    drop({id, segment}) {
+        this.client.drop({id, segment});

```

Comment:
According to spec [this.client.drop](https://github.com/hapijs/catbox#api)  is an async operation. so maybe
```js
async drop({id, segment}) {
    await this.client.drop({id, segment});
    return null;
}
```